server/src-lib/Hasura/RQL/Types/CustomTypes.hs

```diff
@@ -216,28 +217,33 @@ $(J.deriveJSON (J.aesonDrop 3 J.snakeCase) ''CustomTypes)
 emptyCustomTypes :: CustomTypes
 emptyCustomTypes = CustomTypes Nothing Nothing Nothing Nothing
 
-data AnnotatedScalarType
+data AnnotatedScalarType (b :: Backend)
```

Comment:
When we have different kinds of sources, the scalar type can come from any source, so a sum type is more appropriate here:
```
data AnnotatedScalarType = Custom .. | ReusedPG .. | ReusedMySQL ...
```

Reply:
Perhaps the "scalar" option deserves to be renamed to `ASTReusedScalar`, so that the data type becomes:
```
data AnnotatedScalarType (b :: Backend)
  = ASTCustom !ScalarTypeDefinition
  | ASTReusedScalar !G.Name !(ScalarType b)
  deriving (Generic)

Reply:
Done, using existential quantification:
```
data AnnotatedScalarType
  = ASTCustom !ScalarTypeDefinition
  | forall b . (b ~ 'Postgres) => ASTReusedScalar !G.Name !(ScalarType b)
```
(The idea is that the `b ~ 'Postgres` constraint would be relaxed in the future, after we generalized code that uses `ASTReusedScalar`s.)